pkg/providers/labels/labels.go

```diff
@@ -125,6 +126,12 @@ func Get(
 		labels[AKSLabelEBPFDataplane] = consts.NetworkDataplaneCilium
 	}
 
+	if nodeClass.IsLocalDNSEnabled() {
+		labels[AKSLocalDNSStateLabelKey] = "enabled"
+	} else if nodeClass.Spec.LocalDNS != nil && nodeClass.Spec.LocalDNS.Mode != nil && *nodeClass.Spec.LocalDNS.Mode == v1beta1.LocalDNSModeDisabled {
+		labels[AKSLocalDNSStateLabelKey] = "disabled"
```

Comment:
So we only set the disabled label if we _explicitly_ set disabled? What about if it's just not set at all (for example mode is not set at all). Then it's disabled, isn't it?

I would expect we either always set the label to disabled OR leave a very clear comment here explaining why we only set it in this case.